crates/ruma-common/src/push.rs

```diff
@@ -364,23 +368,32 @@ impl From<SimplePushRuleInit> for SimplePushRule {
 // The following trait are needed to be able to make
 // an IndexSet of the type
 
-impl Hash for SimplePushRule {
+impl<T> Hash for SimplePushRule<T>
+where
+    T: Hash,
+{
     fn hash<H: Hasher>(&self, state: &mut H) {
         self.rule_id.hash(state);
     }
 }
 
-impl PartialEq for SimplePushRule {
+impl<T> PartialEq for SimplePushRule<T>
+where
+    T: PartialEq<T>,
+{
     fn eq(&self, other: &Self) -> bool {
         self.rule_id == other.rule_id
     }
 }
 
-impl Eq for SimplePushRule {}
+impl<T> Eq for SimplePushRule<T> where T: PartialEq<T> {}
```

Comment:
This seems wrong, why not `T: Eq`?

Reply:
There's not really a good reason. I guess I don't like when things look too simple :stuck_out_tongue: